server/src/main/java/org/elasticsearch/common/blobstore/OperationPurpose.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.blobstore;
+
+public enum OperationPurpose {
```

Comment:
I see us mentioning stats collection in every method javadoc where this is passed. I think that is only one use out of potentially several. Instead, we could supply that information here and leave the method level javadoc without the use case.